prime-angular-frontend/src/app/modules/site-registration/pages/device-provider-page/device-provider-page-form-state.class.ts

```diff
@@ -0,0 +1,73 @@
+import { FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+import { AbstractFormState } from '@lib/classes/abstract-form-state.class';
+import { FormControlValidators } from '@lib/validators/form-control.validators';
+
+import { IndividualDeviceProvider } from '@registration/shared/models/individual-device-provider.model';
+
+export class DeviceProviderPageFormState extends AbstractFormState<IndividualDeviceProvider[]> {
+  public constructor(
+    private fb: FormBuilder
+  ) {
+    super();
+
+    this.buildForm();
+  }
+
+  public get individualDeviceProviders(): FormArray {
+    return this.formInstance.get('individualDeviceProviders') as FormArray;
+  }
+
+  public get json(): IndividualDeviceProvider[] {
+    if (!this.formInstance) {
+      return;
+    }
+
+    return this.individualDeviceProviders.getRawValue();
+  }
+
+  public patchValue(providers: IndividualDeviceProvider[]): void {
+    if (!this.formInstance || !Array.isArray(providers)) {
+      return;
+    }
+
+    this.individualDeviceProviders.clear(); // Clear out existing indices
+
+    providers.forEach((p: IndividualDeviceProvider) => this.push(p));
+  }
+
+  public at(index: number): FormGroup {
```

Comment:
Since it's `public` it might be worth providing a bit of context to the method name like `individualDeviceProvidersAt(index: number)` or `individualDeviceProviderAt(index: number)`

Reply:
Only use it once in the component so just removed it and call it directly to make it clearer.